### ISSUE TITLE
[Bug] Device Provisioning error

### ISSUE THREAD

**Issue by jonrot1906** [thingsboard/thingsboard#3909]
**Describe the bug**
When using the newly introduced MQTT device API for provisioning my devices by supplying a X.509 certificate, I get a SSL WRONG_VERSION_NUMBER error most of the time. Sometimes it works, but most of the time it does not. This is the described error:
```[Provisioning client] Connecting to ThingsBoard (provisioning client)
Traceback (most recent call last):
  File "prov.py", line 189, in <module>
    provision_client.provision()  # Request provisioned data
  File "prov.py", line 129, in provision
    self.connect(self._host, self._port, 60)
  File "/usr/local/lib/python3.7/dist-packages/paho/mqtt/client.py", line 941, in connect
    return self.reconnect()
  File "/usr/local/lib/python3.7/dist-packages/paho/mqtt/client.py", line 1104, in reconnect
    sock.do_handshake()
  File "/usr/lib/python3.7/ssl.py", line 1117, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1056)
```

MQTT over SSL is configured following the official Thingsboard instructions (I've created a TLS v1.2 server certificate). Also one way MQTT over SSL works. Two way MQTT over SSL brings up the same error when using python and the paho mqtt library to connect, Nodered works like a charm with both access tokens and key/cert.

**Environment**

 * OS:  Raspberry Pi 4 2 GB
 * ThingsBoard: 3.2.0

**Your Server Environment**

* own setup
  * docker deployment
  * ThingsBoard Version: 3.2.0
  * OS Name and Version: Ubuntu 20.04 LTS

**Your Client Environment**

**Your Device**
Raspbery Pi 4 2 GB running Hypriot OS

* Connectivity
  * MQTT

**Comment by jonrot1906** [comment]
Obviously two way ssl over python isn't working for me, so I removed the connect request after the provisioning part in the script as this one throws the error most of the time.
I also used a simple while loop to ensure provisioning without errors (provisioning failed because of ssl errror ==> repeat until provisioning is successful).

I then used two way ssl with Node.js, now it works like a charm.
